externals/kyuubi-spark-monitor/src/main/scala/org/apache/kyuubi/engine/spark/monitor/listener/KyuubiStatementListener.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark.monitor.listener
+
+import org.apache.spark.scheduler._
+
+import org.apache.kyuubi.KyuubiSparkUtils._
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.engine.spark.monitor.KyuubiStatementMonitor
+import org.apache.kyuubi.engine.spark.monitor.entity.KyuubiJobInfo
+
+/**
+ * This listener is used for getting metrics about job, stage, executor and stageExecutor.
+ * It's singleton pattern and we will add it into sparkContext when initialize sparkEngine.
+ */
+class KyuubiStatementListener extends StatsReportListener with Logging{
```

Comment:
Merge KyuubiStatementListener to SparkSQLEngineListener

Reply:
I don't think it's a good idea, even though we add them into sparkContext in the same time.
First, event granularity is different. In KyuubiStatementListener, i don't care about this engine(SparkSQLEngineListener is for engine level), the listener is used for getting metrics about job, stage, executor for each statement.
Second,  i used StatsReportListener as parent class.

Reply:
we need to avoid `StatsReportListener` be inherit twice

Reply:
if you don't care about the engine, what stops you from merging them?

Reply:
In StatsReportListener's code comment, it says "Simple SparkListener that logs a few summary statistics when each stage completes." I think i can use them...

Reply:
In my listener, i don't care about engine.
I think engine and statement should have different listener.

Reply:
why？

Reply:
FYI, StatsReportListener is inherited by SQLOperationListener already, we currently used it for logging in both the operation log  and the engine log, if it is involved here again, we are going to produce completely duplicated logs